### ISSUE TITLE
Not a HASH reference at gottcha_db.pl line 1573.

### ISSUE THREAD

**Issue by playerra** [LANL-Bioinformatics/GOTTCHA#3]
I'm currently stuck at the gottcha_db.pl part, building a custom database, with the following error:

...
Tues 06/02/2015(15:51:17) ------------------------------------------------------------------
Tues 06/02/2015(15:51:17) Importing XML file...done.
Not a HASH reference at gottcha_db.pl line 1573.
Command exited with non-zero status 255

Any idea what that 'Not a HASH reference' means? I checked in the actual code and can't make much sense of it.

Thanks,
Robert

**Comment by poeli** [comment]
It seems that the script can’t find the corresponding taxonomy for given GI number. Can you double check it?

**Comment by playerra** [comment]
You are correct. Looking back at my mkGottchaTaxTree.log I found:

Indexing GIs by TAXID...
    **Warning: GI "300693042" has no TAXID!**
    **Warning: GI "409246494" has no TAXID!**
    **Warning: GI "479208076" has no TAXID!**
and so on...

Suggestions?

**Comment by playerra** [comment]
So when I check the first GI on ncbi it returns:
"Record removed. This RefSeq genome was suppressed because updated RefSeq validation criteria identified problems with the assembly or annotation."

There are 300 or so of these warnings from the mkGottchaTaxTree.log. These GIs that have been warned should not be searched for a TAXID during gottcha_db.pl database build (which resulted in my intial error at the top of this thread), right?

**Comment by poeli** [comment]
Yes, you are right. I will need to remove them before you run the database generating scripts. Thanks.

On Jun 3, 2015, at 12:27 PM, playerra <notifications@github.com<mailto:notifications@github.com>> wrote:


So when I check the first GI on ncbi it returns:
"Record removed. This RefSeq genome was suppressed because updated RefSeq validation criteria identified problems with the assembly or annotation."

There are 300 or so of these warnings from the mkGottchaTaxTree.log. These GIs that have been warned should not be searched for a TAXID during gottcha_db.pl database build (which resulted in my intial error at the top of this thread), right?

**Comment by playerra** [comment]
Paul, thanks for the update.
Could you please clarify from what file I need to remove those 300 GIs? Do they need to be removed from the 'genomes.txt' file containing the path to all the .gbk files of Bacteria and Viruses? Or from the SpeciesTreeGI.dmp that is created? I'm pretty lost here.

**Comment by poeli** [comment]
Hi Robert,

You will need to remove paths of those 300 gbk files from genomes.txt file.

We did do a bunch of curation works on the reference genomes upfront in order to generate correct signatures. So cleaning removed/mistake genomes is required at this moment. The database building scripts could be annoying if the input genomes are directly downloaded from NCBI.

Thanks,
Paul

On Jun 4, 2015, at 11:55 AM, playerra <notifications@github.com<mailto:notifications@github.com>> wrote:


Paul, thanks for the update.
Could you please clarify from what file I need to remove those 300 GIs? Do they need to be removed from the 'genomes.txt' file containing the path to all the .gbk files of Bacteria and Viruses? Or from the SpeciesTreeGI.dmp that is created? I'm pretty lost here.
